uh50_api/uh50_reader.py

```diff
@@ -0,0 +1,69 @@
+""" 
+Reads raw UH50 response data from the UH50 unit. 
+To test the connection use validate, which will return the model name.
+"""
+import serial
+from serial import Serial
+
+
+class UH50Reader:
+    def __init__(self, port) -> None:
+        self._port = port
+        self.model: str
+
+    def validate(self) -> str:
+        "Open connection to the UH50 and get the model name, thereby validating the connection"
+        with self._connect_serial() as conn:
+            self._wake_up(conn)
+        return self.model
+
+    def read(self) -> str:
+        "Reads the UH50 on the specified port, returning the full string"
+        with self._connect_serial() as conn:
+            return self._get_data(conn)
+
+    def _connect_serial(self) -> Serial:
+        "Make the connection to the serial device"
+        return serial.Serial(
+            self._port,
+            baudrate=300,
+            bytesize=serial.SEVENBITS,
+            parity=serial.PARITY_EVEN,
+            stopbits=serial.STOPBITS_TWO,
+            timeout=1,
+            xonxoff=0,
+            rtscts=0,
+        )
+
+    def _wake_up(self, conn) -> str:
+        "Wake up the device and get the model number"
+        # Waking up should be done at 300 baud
+        # Sending /?!
+        conn.write(
+            b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x2F\x3F\x21\x0D\x0A"
+        )
+        ir_command = "/?!\x0D\x0A"
+        conn.write(ir_command.encode("utf-8"))
+        conn.flush()
+        # Read at 300 baud, this gives us the typenr
+
+        # checking if we can read the model (eg. 'LUGCUH50')
+        self.model = conn.readline().decode("utf-8")[1:9]
+        if not (self.model):
+            raise Exception("No model could be read")
+        return

```

Comment:
Ik zou van model geen field maken, maar m hier returnen. Zo houdt je de scope van je variabelen zo klein mogelijk. Dat is een goed idee want shared state is over het algemeen niet zo'n goed idee en in dit geval dus ook niet nodig.

Reply:
Zie ook mij comment hiervoor. Het is wel gebruikelijk om die data aan de client kant te bewaren. Dat is namelijk een ander "concern" dan waar die service voor is.

Reply:
Bekijk hier ook eens de aanroepende kant.

Reply:
Dit is de aanroepende kant: https://github.com/vpathuis/core/tree/uh50_heat_meter/homeassistant/components/uh50
Daar heb ik niet eindeloos vrijheid om alles te doen, hoewel er wel veel kan. Maar wat ik bij andere componenten zie gebeuren is wel die combi service/state, als ik me niet vergis.

Reply:
Ik heb het aan de voorkant kunnen aanpassen dat ik het response field niet nodig heb. Kleine aanpassing eigenlijk. Ik ga daar nu even vanuit, maar wel goed om nog even over te bellen.

Reply:
Ik bedoelde eigenlijk regel 17 in deze file. Daar heb ik ook een comment bij geplaatst. Als ik het linkje volg, zie ik geen call naar deze class meen ik. Misschien ook iets wat je me ff kunt laten zien?

Reply:
Ik heb de commit even valide voor nu, dus als je 'm nog eens bekijkt ziet het er anders uit. 
Hier wordt het model opgehaald bij het aanmaken van het device: https://github.com/vpathuis/core/blob/8b9f1ba6259cabe1ceb2dd30e0035c1fd23d68ce/homeassistant/components/uh50/config_flow.py#L109
Hier wordt bij initialisatie de service bewaard om later aan te kunnen roepen: https://github.com/vpathuis/core/blob/8b9f1ba6259cabe1ceb2dd30e0035c1fd23d68ce/homeassistant/components/uh50/__init__.py#L28
En in deze file gebeurt het echte werk, resultaten ophalen: https://github.com/vpathuis/core/blob/8b9f1ba6259cabe1ceb2dd30e0035c1fd23d68ce/homeassistant/components/uh50/sensor.py#L33
En toepassen: https://github.com/vpathuis/core/blob/8b9f1ba6259cabe1ceb2dd30e0035c1fd23d68ce/homeassistant/components/uh50/sensor.py#L90